ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -337,9 +337,10 @@ private String visitBooleanComparisonExpression(ComparisonExpression.Type type)
           exprFormat += visitStringComparisonExpression(node.getType());
           break;
         case MAP:
-          throw new KsqlException("Cannot compare MAP values");
         case ARRAY:
-          throw new KsqlException("Cannot compare ARRAY values");
+        case STRUCT:
+          throw new KsqlException(
```

Comment:
should this be the `default` case?
Also, why is the current `default`
`exprFormat += visitScalarComparisonExpression(node.getType());` ?
That seems a bit wrong